### PR TITLE
Align PR summary table to be more readable in its raw form

### PR DESCRIPTION
**Pull request by Kobzol**
Aligns the columns to the same width (more or less), to be more readable in plain text. Let me know what you think before I update all the tests :)

Before:
```
| | Regressions 😿 <br />(primary) | Regressions 😿 <br />(secondary) | Improvements 🎉 <br />(primary) | Improvements 🎉 <br />(secondary) | All 😿 🎉 <br />(primary) |
|:---:|:---:|:---:|:---:|:---:|:---:|
| count[^1] | 2 | 0 | 2 | 0 | 4 |
| mean[^2] | 150.0% | N/A | -62.5% | N/A | 43.8% |
| max | 200.0% | N/A | -75.0% | N/A | 200.0% |

[^1]: *number of relevant changes*
[^2]: *the arithmetic mean of the percent change*
```

After:
```
|            | Regressions 😿 <br />(primary) | Regressions 😿 <br />(secondary) | Improvements 🎉 <br />(primary) | Improvements 🎉 <br />(secondary) | All 😿 🎉 <br />(primary) |
|:----------:|:------------------------------:|:--------------------------------:|:-------------------------------:|:---------------------------------:|:------------------------:|
| count[^1]  | 2                              | 0                                | 2                               | 0                                 | 4                        |
| mean[^2]   | 150.0%                         | N/A                              | -62.5%                          | N/A                               | 43.8%                    |
| max        | 200.0%                         | N/A                              | -75.0%                          | N/A                               | 200.0%                   |

[^1]: *number of relevant changes*
[^2]: *the arithmetic mean of the percent change*
```

Fixes: https://github.com/rust-lang/rustc-perf/issues/1310

### CODE REVIEW

**Comment by klensy** [comment]
Maybe remove emoji from column titles?

When there was not a table (in previous version of reports), but lines with emoji, was used only a few ones when appropriate, but now there a lot of it unconditionally.

**Comment by Kobzol** [comment]
I realize that emojis are mostly useless for plain text, but at least for me it's quite useful in GitHub comments, as it allows me to quickly identify which columns are the regressions and which are the improvements.

File: site/src/comparison.rs [hunk 1]
```diff
@@ -411,47 +411,70 @@ pub fn write_summary_table(
...
+        write!(result, "| {} ", column).unwrap();
+    }
+    result.push_str("|\n");
+    for &count in &counts {
+        write!(result, "|:{}:", "-".repeat(count)).unwrap();
+    }
+    result.push_str("|\n");
+
+    let mut render_row = |row: Vec<String>| {
+        assert_eq!(row.len(), columns.len());
```
Comment (rylev) [inline site/src/comparison.rs:444]: Nit: let's make this a debug assert. There's no need to panic in prod.

File: site/src/comparison.rs [hunk 2]
```diff
@@ -411,47 +411,70 @@ pub fn write_summary_table(
...
+        .map(|s| {
+            // Unicode emojis usually have width equal to ~2 chars (maybe? :) ).
+            let count = s.chars().count();
+            if s.chars().any(|c| !c.is_whitespace()) {
+                count + 1
+            } else {
+                count
+            }
+        })
+        .collect();
```
Comment (rylev) [inline site/src/comparison.rs:433]: Do we need to collect this since it will just be used in the call to `zip`?

**Review by rylev — APPROVED** [review]
Looks good. Just need to fix the CI breakage.

File: site/src/comparison.rs (hunk 2 above)
Comment (Kobzol) [reply site/src/comparison.rs:433]: It's used in two places, so it has to be collected into a `Vec` (afaik you can't really reuse an iterator in a generic way?).

**Comment by Kobzol** [comment]
Fixed tests and changed the assert to `debug_assert`.

**Pull request merged**